graylog2-server/src/main/java/org/graylog/events/processor/EventProcessorExecutionJob.java

```diff
@@ -122,8 +126,21 @@ public JobTriggerUpdate execute(JobExecutionContext ctx) throws JobExecutionExce
             // we are using in Elasticsearch is only using millisecond precision. ("date") Once we switch to a
             // different date type with nanosecond precision (e.g. "date_nanos"), this workaround will break and we
             // will miss messages.
-            final DateTime nextTo = to.plus(config.processingHopSize());
-            final DateTime nextFrom = nextTo.minus(config.processingWindowSize()).plusMillis(1);
+            DateTime nextTo = to.plus(config.processingHopSize());
+            DateTime nextFrom = nextTo.minus(config.processingWindowSize()).plusMillis(1);
+
+            // If the event processor is catching up on old data (e.g. the server was shut down for a significant time),
+            // we can switch to a bigger scheduling window: `processingCatchUpWindowSize`.
+            // If enabled, it will ignore the processingHopSize and will schedule a tumbling window timerange of `processingCatchUpWindowSize` chunks.
+            final long catchUpSize = configurationProvider.get().eventCatchupWindow();
+            if (catchUpSize > 0 && catchUpSize > config.processingWindowSize() && to.plus(catchUpSize).isBefore(now)) {
+                final long chunkCount = catchUpSize / config.processingWindowSize();
+
+                // Align to multiples of the processingWindowSize
+                nextTo = to.plus(config.processingWindowSize() * chunkCount);
+                LOG.debug("eventproc <{}> is catching up on old data. Combining {} search windows with catchUpWindowSize={}ms: from={} to={}",
```

Comment:
@bernd I'm wondering if we should make this `LOG.info` what do you think?

Reply:
@mpfz0r Sounds good to me! But please replace `eventproc` with `Event processor` if we show it to users. :slightly_smiling_face:

Reply:
Cool. Did that already :-)